lib/timeline/component/Group.js

```diff
@@ -212,119 +212,160 @@ Group.prototype.getLabelWidth = function() {
  * @param {{start: number, end: number}} range
  * @param {{item: {horizontal: number, vertical: number}, axis: number}} margin
  * @param {boolean} [forceRestack=false]  Force restacking of all items
- * @return {boolean} Returns true if the group is resized
+ * @param {boolean} [returnQueue=false]  return the queue or the result 
+ * @return {boolean} Returns true if the group is resized or the redraw queue if returnQueue=true
  */
-Group.prototype.redraw = function(range, margin, forceRestack) {
-  var resized = false;
- 
-  // force recalculation of the height of the items when the marker height changed
-  // (due to the Timeline being attached to the DOM or changed from display:none to visible)
-  var markerHeight = this.dom.marker.clientHeight;
-  if (markerHeight != this.lastMarkerHeight) {
-    this.lastMarkerHeight = markerHeight;
-    util.forEach(this.items, function (item) {
-      item.dirty = true;
-      if (item.displayed) item.redraw();
-    });
-
-    forceRestack = true;
-  }  
-  
-  // recalculate the height of the subgroups
-  this._calculateSubGroupHeights(margin);
-
-  // calculate actual size and position
-  var foreground = this.dom.foreground;
-  this.top = foreground.offsetTop;
-  this.right = foreground.offsetLeft;
-  this.width = foreground.offsetWidth;
-
-  var lastIsVisible = this.isVisible;
-  this.isVisible = this._isGroupVisible(range, margin);
-  
-  var restack = forceRestack || this.stackDirty || (this.isVisible && !lastIsVisible);
-
-  this._updateSubgroupsSizes();
-  // if restacking, reposition visible items vertically 
-  if(restack) {
-    if (typeof this.itemSet.options.order === 'function') {
-      // a custom order function
-      // brute force restack of all items
-
-      // show all items
-      var me = this;
-      var limitSize = false;
-      util.forEach(this.items, function (item) {
-        if (!item.dom) { // If this item has never been displayed then the dom property will not be defined, this means we need to measure the size
-          item.redraw();
-          me.visibleItems.push(item);
+Group.prototype.redraw = function(range, margin, forceRestack, returnQueue) {
+  var queue = [];
```

Comment:
If pure performance is the concern, I think it would be more readable slightly more performant (maybe a micro optimization?) to have all of these anonymous functions defined at a higher level, and then simply:

```javascript
var queue = [recalcHeight(this), recalcSubGroupHeights(this), calcSizeAndPosition, etc...];
if (returnQueue) {
    return queue;
} else {
    var result;
    queue.forEach(function (fn) {
        result = fn(this);
    });
    return result;
}
...
```

Reply:
I agree but I was planning doing it in another PR

Reply:
How often does redraw get called?

Reply:
every zoom/scroll/drag - alot

Reply:
Ok, can you include a reference to the follow-up issue? Just want to make sure it doesn't get lost.

Reply:
I can include it here if you think it's really important. 
Wait with this PR if so

Reply:
I think it's worth doing, given how often this is used.